imports/ui/ledger/ledger.js

```diff
@@ -396,9 +383,71 @@ export class Ledger {
             }
         };
 
-        txSkeleton.value.msg = [txMsg];
+        return Ledger.createSkeleton(txContext, [txMsg]);
+    }
+
+    static createSubmitProposal(
+        txContext,
+        title,
+        description,
+        deposit
+    ) {
+        const txMsg = {
+            type: 'cosmos-sdk/MsgSubmitProposal',
+            value: {
+                content: {
+                    type: "cosmos-sdk/TextProposal",
```

Comment:
There are 3 types of proposal `text`, `parameter_change` and `software_upgrade`.